packages/flutter/test/material/popup_menu_theme_test.dart

```diff
@@ -0,0 +1,282 @@
+      theme: ThemeData(popupMenuTheme: popupMenuTheme),
+      key: popupButtonApp,
+      home: Material(
+        child: Column(
+          children: <Widget>[
+            PopupMenuButton<void>(
+              key: popupButtonKey,
+              itemBuilder: (BuildContext context) {
+                return <PopupMenuEntry<Object>>[
+                  PopupMenuItem<void>(
+                    key: popupItemKey,
+                    child: const Text('Example'),
+                  ),
+                ];
+              },
+            ),
+          ],
+        ),
+      ),
+    ));
+
+    await tester.tap(find.byKey(popupButtonKey));
+    await tester.pumpAndSettle();
+
+    final Material button = tester.widget<Material>(
+      find.descendant(
+        of: find.byKey(popupButtonApp),
+        matching: find.byType(Material),
+      ).last,
+    );
+    expect(button.color, popupMenuTheme.color);
+    expect(button.shape, popupMenuTheme.shape);
+    expect(button.elevation, popupMenuTheme.elevation);
+
+    final AnimatedDefaultTextStyle text = tester.widget<AnimatedDefaultTextStyle>(
+      find.descendant(
+        of: find.byKey(popupItemKey),
+        matching: find.byType(AnimatedDefaultTextStyle),
+      ).last,
+    );
+    expect(text.style, popupMenuTheme.textStyle);
+  });
+
+  testWidgets('PopupMenuEntry widget properties take priority over theme', (WidgetTester tester) async {
+    final PopupMenuThemeData popupMenuTheme = _popupMenuTheme();
+    final Key popupButtonKey = UniqueKey();
+    final Key popupButtonApp = UniqueKey();
+    final Key popupItemKey = UniqueKey();
+
+    const Color color = Colors.purple;
+    const ShapeBorder shape = RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(9.0)),
+    );
+    const double elevation = 7.0;
+    const TextStyle textStyle = TextStyle(color: Color(0x00000000), textBaseline: TextBaseline.alphabetic);
+
+    await tester.pumpWidget(MaterialApp(
+      theme: ThemeData(popupMenuTheme: popupMenuTheme),
+      key: popupButtonApp,
+      home: Material(
+        child: Column(
+          children: <Widget>[
+            PopupMenuButton<void>(
+              key: popupButtonKey,
+              elevation: elevation,
+              color: color,
+              shape: shape,
+              itemBuilder: (BuildContext context) {
+                return <PopupMenuEntry<Object>>[
+                  PopupMenuItem<void>(
+                    key: popupItemKey,
+                    textStyle: textStyle,
+                    child: const Text('Example'),
+                  ),
+                ];
+              },
+            ),
+          ],
+        ),
+      ),
+    ));
+
+    await tester.tap(find.byKey(popupButtonKey));
+    await tester.pumpAndSettle();
+
+    final Material button = tester.widget<Material>(
+      find.descendant(
+        of: find.byKey(popupButtonApp),
+        matching: find.byType(Material),
+      ).last,
```

Comment:
It's not clear what the ".last" material widget is. Maybe add a comment (here and below).

Reply:
Done